src/ebtree/src/ebtree.erl

```diff
@@ -282,6 +283,9 @@ reduce(Db, #tree{} = Tree, StartKey, EndKey, Options) ->
     do_reduce(Tree, MapValues, ReduceValues).
 
 
+do_reduce(#tree{} = Tree, [], []) ->
+    reduce_values(Tree, [], false);
```

Comment:
shouldn't that be `true`? the lack of map values is the signal we're working with reductions.

Reply:
We're not working with reductions though. We're reducing an empty list of key/value pairs. Plainly the empty list from the second argument is the same empty list being passed to reduce_values.

Also, I think most folks writing reduce functions would do better to think of it as "reduce empty list of key/values" vs "rereduce empty list of reduce values that might need to be guarded against for the normal case when you actually have reduce values".

Reply:
kk